pkg/archive/compression/compression.go

```diff
@@ -0,0 +1,222 @@
+package compression
+
+import (
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime/debug"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/containerd/log"
+	"github.com/docker/docker/pkg/pools"
+	"github.com/klauspost/compress/zstd"
+)
+
+// Compression is the state represents if compressed or not.
+type Compression int
+
+const (
+	Uncompressed Compression = 0 // Uncompressed represents the uncompressed.
```

Comment:
How about `compression.None` instead of `compression.Uncompressed`?

```suggestion
	None Compression = 0 // None represents the uncompressed.
```

Reply:
Ah, yes, that works for me; nice and short.